packages/core/src/types.ts

```diff
@@ -18,12 +20,44 @@ export type ComponentSchema = {
 
 export type PageSchema = {
   name: string;
-  path: string;
+  navigation: {
+    path: string;
+    tab?: {
+      label: string;
+      icon: string;
+    };
+  };
   children: ComponentSchema[];
 };
 
+export type Platform = "web" | "native" | "webcomponent";
+
 export type AppSchema = {
-  name: string;
+  batiq: string;
+  platform: Platform;
+  info: Record<string, any> & {
+    name: string;
+    description?: string;
+    version?: string;
+    tags?: {
+      name: string;
+      description?: string;
+    }[];
+  };
+  prefixes: string[];
```

Comment:
Looks like prefixes is intended for linking base path, IMHO it's better to add in app `config`:
```json
{
  "info": {},
  "config": {
    "LINK_PREFIXES": []
  }
}
```